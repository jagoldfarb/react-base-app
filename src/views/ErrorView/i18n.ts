import i18next from 'i18next';
import { locales } from 'configs/i18n';

i18next.addResources(locales.EN, 'ERROR_VIEW', {
  PAGE_NOT_FOUND: 'Page not found',
});
