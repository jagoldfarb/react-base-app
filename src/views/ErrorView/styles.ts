import styled from 'styled-components';

import { COLORS } from 'helpers/enums/colors';

export const ErrorContainer = styled.div`
  background-color: ${COLORS.white};
`;
