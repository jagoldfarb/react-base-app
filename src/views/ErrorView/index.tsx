import React, { FunctionComponent } from 'react';
import i18next from 'i18next';

import LayoutContainer from 'containers/LayoutContainer';

import { ErrorContainer } from './styles';
import './i18n';

const ErrorView: FunctionComponent = () => (
  <LayoutContainer>
    <ErrorContainer data-testid="error-view">
      <h1>{i18next.t('ERROR_VIEW:PAGE_NOT_FOUND')}</h1>
    </ErrorContainer>
  </LayoutContainer>
);

export default ErrorView;
