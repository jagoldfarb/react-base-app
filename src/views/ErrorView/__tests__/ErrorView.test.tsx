import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import store from 'redux/store';

import ErrorView from '..';

describe('Error view', () => {
  it('Should render the error page', () => {
    const { asFragment, getByTestId, getByText } = render(
      <Provider store={store}>
        <MemoryRouter>
          <ErrorView />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('error-view')).toBeDefined();
    expect(getByText('Page not found')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
