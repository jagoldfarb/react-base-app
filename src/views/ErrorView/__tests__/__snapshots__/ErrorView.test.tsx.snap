// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error view Should render the error page 1`] = `
<DocumentFragment>
  <div
    class="sc-fFubgz ghXXhE"
  >
    <div
      class="sc-gsTCUz isGlRi"
      data-testid="header"
    >
      <div
        class="sc-dlfnbm sofHW"
      >
        <div
          class="sc-bdfBwQ fajWDP"
          data-testid="logo"
        >
          <a
            href="/home"
          >
            Logo
          </a>
        </div>
        <div
          class="sc-hKgILt MKpTd"
          data-testid="menu-overlay"
        />
        <div
          class="sc-eCssSg laWbwL"
          data-testid="menu-content"
        >
          <div
            class="sc-jSgupP dgpcrU"
          >
            <svg
              class="sc-gKsewC gPMeTF"
            >
              open-menu.svg
            </svg>
          </div>
          <ul
            class="sc-fubCfw cmWXKT"
          >
            <li>
              <a
                data-testid="nav-link-1"
                href="/link-1"
              >
                Link 1
              </a>
            </li>
            <li>
              <a
                data-testid="nav-link-2"
                href="/link-2"
              >
                Link 2
              </a>
            </li>
            <li>
              <a
                data-testid="nav-link-3"
                href="/link-3"
              >
                Link 3
              </a>
            </li>
            <li>
              <a
                data-testid="nav-link-signup"
                href="/signup"
              >
                Sign Up
              </a>
            </li>
            <li>
              <a
                data-testid="nav-link-login"
                href="/login"
              >
                Log In
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="sc-bkzZxe eAnuUR"
    >
      <div
        class="sc-idOhPF bcrWPj"
        data-testid="error-view"
      >
        <h1>
          Page not found
        </h1>
      </div>
    </div>
    <footer
      class="sc-pFZIQ bJlbpl"
      data-testid="footer"
    >
      <div
        class="sc-jrAGrp duSMMQ"
      >
        <div
          class="sc-kEjbxe eAqEvN"
        >
          <div
            class="sc-bdfBwQ kCPbJH"
            data-testid="logo"
          >
            <a
              href="/home"
            >
              Logo
            </a>
          </div>
          <p
            class="sc-iqHYGH bczUIk"
          >
            Â© Copyright 2021
          </p>
        </div>
        <div
          class="sc-crrsfI hIsiGn"
        >
          <ul
            class="sc-dQppl jVPArR"
            data-testid="footer-links-container"
          >
            <li>
              <a
                href="/link-1"
              >
                Link 1
              </a>
            </li>
            <li>
              <a
                href="/link-2"
              >
                Link 2
              </a>
            </li>
            <li>
              <a
                href="/link-3"
              >
                Link 3
              </a>
            </li>
          </ul>
          <div
            class="sc-bqyKva botMR"
            data-testid="footer-social-medias-container"
          >
            <a
              href="https://www.linkedin.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="sc-fodVxV kVUtmk"
              >
                LinkedInIcon.svg
              </svg>
            </a>
            <a
              href="https://www.instagram.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="sc-hBEYos begnOL"
              >
                InstagramIcon.svg
              </svg>
            </a>
            <a
              href="https://www.facebook.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="sc-kstrdz BPBfR"
              >
                FacebookIcon.svg
              </svg>
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
